extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -55,6 +55,12 @@ class AmpSocialShare extends AMP.BaseElement {
   buildCallback() {
     const typeAttr = user().assert(this.element.getAttribute('type'),
         'The type attribute is required. %s', this.element);
+    const isTypeSupported = typeAttr !== 'system' ||
+        navigator.share !== undefined;
+    if (!isTypeSupported) {
```

Comment:
Actually, lets just make this special
`if (typeAttr == 'system' && !navigator.share)`

No need to overgeneralize if there is only one case. And then please add a comment.